### DISCUSSION TITLE
Allow for mania LN tail to have TopCentre origin

### DISCUSSION THREAD

**Discussion by longnguyen2004** [ppy/osu#26470] [Ideas] [answered]
Currently, the LN tail has an origin of BottomCentre, same as the head.

This creates a problem if you want a LN tail that doesn't extend into the next note, to increase readability in dense LN sections. The community has come up with a workaround, which is the 40000px body texture. This is a huge hack, which goes against how GPU works, and even with the best effort support that we currently have in osu!lazer, it still doesn't match stable 1:1.

Instead of that, I propose we allow the LN tail to have an origin of TopCentre (likewise BottomCentre for upscrolling), which will allow the LN tail to protrude into the body, solving the majority of cases that currently requires the huge texture hack. To maintain compatibility with existing skins, a new skin setting can be added that controls whether the tail is top aligned or not.

Is this a viable enough solution?

**Comment by Adrriii** [comment]
This would definitely go a long way, and fits with what I posted previously (but this proposition is much cleaner)

**Comment by peppy** [comment]
I think this can be implemented as a `skin.ini` setting as a stopgap measure.

**Reply by frenzibyte** [reply, marked as answer]
https://github.com/ppy/osu/pull/27726
